Contest_Archive_Format.md

```diff
@@ -16,708 +17,166 @@ disk, including:
-
-- The `time_limit` element is required to be a `number` that is an
-  integer multiple of `0.001` in the Contest API. 
-
-#### Examples
-
-```json
-{
-  "id": "asteroids",
-  "uuid": "cb1c4b77-b203-4943-a13f-9b89dec1ac11",
-  "label": "A",
-  "name": "Asteroid Rangers",
-  "ordinal": 1,
-  "rgb":"#00f",
-  "color": "blue",
-  "time_limit": 2,
-  "test_data_count": 10
-}
-```
-
-### Group object
-
-| Name    | Type    | Required | Description |
-| :------ | :------ | :------- | :---------- |
-| id      | ID      | yes      | Identifier of the group. |
-| icpc_id | string  | no       | External identifier from ICPC CMS. |
-| name    | string  | yes      | Name of the group. |
-| type    | string  | no       | Type of this group. |
-| hidden  | boolean | no       | If group should be hidden from scoreboard. Defaults to false if missing. |
-
-#### Differences from Contest API
-
-None.
-
-#### Examples
-
-```json
-{
-  "id": "asia",
-  "icpc_id": "7593",
-  "name": "Asia"
-}
+A package contains information regarding a single contest (corresponding to
+the `/contests/<id>/*` endpoints of the API). The API can contain information
+for several contests, to store all information a package per contest would be
+needed.
+
+Information in the API is always either in JSON format, [NDJSON]
+(contest_api#event-feed) format, or linked using a [file reference]
+(contest_api#json-attribute-types) JSON object.
+
+- The JSON returned from the endpoint `/` is stored as
+  `api.json`.
+- The JSON returned from the endpoint `/contests/<id>` is stored as
+  `contest.json`. (Notice the singular form).
+- The JSON returned from the endpoint `/contests/<id>/<endpoint>` is stored as
+  `<endpoint>.json`.
+- The NDJSON returned from the endpoint `/contests/<id>/<endpoint>` is stored as
+  `<endpoint>.ndjson`. (The only such endpoint is `event-feed`) 
+
+
+Files referenced to in `contest.json` are stored as `contest/<filename>`, and
+files referenced to in `<endpoint>.json` as `<endpoint>/<id>/<filename>`,
+where:
+- `<id>` is the ID of the endpoint object the reference is in.
+- `<filename>` is the filename specified in the file reference object.
+
+Note that the API specification requires that filenames are unique within
+endpoint objects, so this is always possible.
+
+The href in the file reference may not be valid and should be ignored, but it
+does not have to be removed.
+
+Optionally one could create a Shallow Archive by not storing the files, but
+then the hrefs must be valid. This could be useful in some cases where the
+size of the archive matters.
+
+In some cases it could make sense to merge multiple API sources (of the same
+contest) in a single archive. One example of this would be a contest that was
+running with a primary and [shadow](ccs_system_requirements#shadow-mode) CCS.
+Typically in such cases, most of the data will be identical (or at least the
+differences irrelevant), so only the data that differs (in relevant ways)
```

Comment:
```suggestion
differences should be irrelevant), so only the data that differs (in relevant ways)
```

Reply:
I don't think "should" is right here. Could "would" work? Or could we really not keep it as is?

Reply:
The phrase is missing a verb. How about "are" instead of "should be"?

Reply:
In that case:
```
most of the data is identical (or at least the differences are irrelevant),
```

I.e. change "will be" to "is" and add the "are". 

This feels better than the original to me. Would it work?

Reply:
I did that.